controllers/component_build_controller.go

```diff
@@ -198,71 +255,288 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, nil
 }
 
-func (r *ComponentBuildReconciler) EnsurePACRepository(ctx context.Context, component appstudiov1alpha1.Component) error {
-	repository := gitops.GeneratePACRepository(component)
-	existingRepository := &pacv1alpha1.Repository{}
-	if err := r.Client.Get(ctx, types.NamespacedName{Name: repository.Name, Namespace: repository.Namespace}, existingRepository); err != nil {
+func (r *ComponentBuildReconciler) getPaCRoutePublicUrl(ctx context.Context) (string, error) {
+	pacWebhookRoute := &routev1.Route{}
+	pacWebhookRouteKey := types.NamespacedName{Namespace: pipelinesAsCodeNamespace, Name: pipelinesAsCodeRouteName}
+	if err := r.Client.Get(ctx, pacWebhookRouteKey, pacWebhookRoute); err != nil {
+		r.Log.Error(err, "failed to get Pipelines as Code route")
```

Comment:
probably intermittent or should we issue event ?

Reply:
I don't think so. We can fail to get the route if PaC is not installed or not ready yet. In that case just retrying should help in waiting for PaC readiness.